frontend/src/visualization/filtering/FilterEntityType.tsx

```diff
@@ -66,45 +67,33 @@ function fetchEdgeTypeFilterModel(
   return filterService.getEdgeTypeFilterModel(edgeName, cancellation);
 }
 
-const EntityFilterElement = (props: {
+const FilterEntityType = (props: {
   backgroundColor: string;
-  name: string;
+  type: string;
   entity: 'node' | 'edge';
-  setFilterQuery: (
-    condition: MatchAllCondition | OfTypeCondition | null
-  ) => void; // TODO: Rename
 }): JSX.Element => {
   const classes = useStyles();
+  const { backgroundColor, type, entity } = props;
 
   // Indicates if filter-dialog is opened.
   const [filterOpen, setFilterOpen] = React.useState(false);
   const [boxShadow, setBoxShadow] = React.useState('None');
   const isActive = React.useRef(false);
 
-  // TODO: Rename
-  const dialogConditionRef = React.useRef<MatchAllCondition | null>(null);
-
-  const { backgroundColor, name, entity, setFilterQuery } = props;
-
   const filterService = useService(FilterService, null);
 
-  // TODO: Rename
-  function updateFilterQuery(): void {
-    const propertiesCondition = dialogConditionRef.current;
+  const [filterQuery, setFilterQuery] = useState<FilterQuery>({});
 
-    if (!isActive.current) {
-      setFilterQuery(null);
-      return;
-    }
+  const filterQueryStore = useService<FilterQueryStore>(FilterQueryStore);
+  const filterStateStore = useService<FilterStateStore>(FilterStateStore);
 
-    const ofTypeCondition = OfTypeCondition(name);
-
-    if (propertiesCondition === null) {
-      setFilterQuery(ofTypeCondition);
-    } else {
-      setFilterQuery(MatchAllCondition(ofTypeCondition, propertiesCondition));
-    }
-  }
+  useObservable(
+    filterQueryStore.getState().pipe(
+      tap((filterQueryFromStore: FilterQuery) => {
+        setFilterQuery(filterQueryFromStore);
+      })
+    )
+  );
```

Comment:
[Optional improvement]
I have updated `useObservable` so that you don't need the extra `useState<FilterQuery>(...)`
```suggestion
  const filterQuery = useObservable(filterQueryStore.getState());
```